test/url_test.rb

```diff
@@ -400,6 +400,15 @@ def test_basic_auth_user_with_password
       assert_equal "Basic #{Base64.encode64("a b:1 + 2").rstrip}", u.basic_auth
     end
 
+    def test_basic_auth_user_with_non_encoded_password
+      u = Url.parse("http://a%20b:MxYut8Rj8tQi6%3DwNf.miTxf%3Eq49%3F%2Cf%40v" \
+                    "QX8og3YT%3Fs.%5D8L3h9)@foo.com")
+      assert_equal "a b", u.user
+      assert_equal "MxYut8Rj8tQi6=wNf.miTxf>q49?,f@vQX8og3YT?s.]8L3h9)", u.password
+      assert_equal "Basic #{Base64.encode64("a b:MxYut8Rj8tQi6=wNf.miTxf>q49?" \
+                                            ",f@vQX8og3YT?s.]8L3h9)").rstrip.gsub(/\n/, '')}", u.basic_auth
```

Comment:
What do you think about just slapping the expected token `YSBiOk14WXV0OFJqOHRRaTY9d05mLm1pVHhmPnE0OT8sZkB2UVg4b2czWVQ/cy5dOEwzaDkp` in there, instead of replicating the logic from the method?